src/ape/api/providers.py

```diff
@@ -556,10 +556,25 @@ def get_events(self, **filter_params) -> Iterator[dict]:
             Iterator[dict]: A dictionary of events.
         """
 
+    def snapshot(self) -> SnapshotID:
```

Comment:
We just need to remember to add new test API methods here.
I think `mypy` will actually help us there, so I am less worried than I was, and this solution is cleaner than the post init idea.